src/package/components/button/Button.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+
+const preventDefault = (callback) => (e) => {
+  e.preventDefault();
+  if(callback) {
+    callback(e);
+  }
+};
+
+export default ({ children, id, onClick, className, style, type }) => {
+  const buttonType = type || 'button'; // Make button default so that when used in forms, "submit" does not become default
+  return (
+    <button
+      id={id}
+      style={style}
+      onClick={preventDefault(onClick)}
```

Comment:
Instead of creating a function to pass a **callback**, may it will be better to check it directly on the prop?
What do you think something like:

```js
<button
  onClick={e => {
    onClick && onClick.constructor === Function && onClick(e) // <~ pass the event object
  }}
/>
```

And, instead of using directly the `e.preventDefault` on the componente, as we passing the event object to the `onClick` function, may would be better to prevent it on the function that we are passing to the component, what do you think?

Something like:

```jsx
// using the Button component, e.preventDefault in onClick prop
import { Button } from '../components/button'

const ButtonExample = (props) => (
  <Button
    {...props}
    onClick={e => {
      e.preventDefault()
      //
      // Do what you need to do
      // 
    }}
  />
)
```